### PR TITLE
Remove CurrentUserLoader and CurrentUserInterface

### PR DESCRIPTION
**Pull request by fraxachun**
Sets the basis for https://github.com/vivid-planet/comet/pull/1684

Removes the unnecessary layer between auth-strategies and the CurrentUser-object. Since the UserPermissionsModule is mandatory by any means it is allowed for it to provide the one and only CurrentUser-DTO.

This change is mainly internal facing. However, it could be breaking if a Comet 6 web uses a custom CurrentUserLoader (which was discouraged but possible) or is messing around with the auth-strategies.

Update: the latest commits which remove the CurrentUserInterface require to replace it in the application (mainly when using @GetCurrentUser(). Should we allow this potentially breaking change in 6.x or export a CurrentUserInterface for legacy reasons?

### CODE REVIEW

File: packages/api/cms-api/src/user-permissions/dto/user.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { Field, ObjectType } from "@nestjs/graphql";
+import { CurrentUserInterface } from "src/auth/current-user/current-user";
 
 @ObjectType()
-export class User {
+export class User implements CurrentUserInterface {
```
Comment (nsams) [inline packages/api/cms-api/src/user-permissions/dto/user.ts:5]: that doesn't seem correct, User implementing CurrentUsterInterface. Shouldn't the interface be completely deleted?

**Comment by nsams** [comment]
> Should we allow this potentially breaking change in 6.x or export a CurrentUserInterface for legacy reasons?

I'd search thru existing v6 projects if it is used

File: docs/docs/migration/migration-from-v5-to-v6.md [hunk 2]
```diff
@@ -54,14 +52,28 @@ It automatically installs the new versions of all `@comet` libraries, runs an ES
...
+
     Change imports of removed classes
 
     ```diff
     - import { CurrentUser, CurrentUserLoader } from "@src/auth/current-user";
     + import { CurrentUser } from "@comet/cms-api";
     ```
 
-    It shouldn't be necessary to override these classes anymore. However, if you really need it, provide the CurrentUserLoader with `CURRENT_USER_LOADER`.
+    Replace occurences of CurrentUserInterface
```
Comment (johnnyomair) [inline docs/docs/migration/migration-from-v5-to-v6.md:69]:
```suggestion
    Replace occurrences of CurrentUserInterface
```

File: .changeset/plenty-humans-grow.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@comet/cms-api": minor
+---
+
+Remove CurrentUserLoader and CurrentUserInterface completely
```
Comment (johnnyomair) [inline .changeset/plenty-humans-grow.md:5]:
```suggestion
Remove `CurrentUserLoader` and `CurrentUserInterface`

Overriding the the current user in the application isn't supported anymore when using the new `UserPermissionsModule`, which provides the current user DTO itself.
```

I'd expand the the changeset with a bit more information.

**Review by johnnyomair — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Pull request merged**